test/rules.bzl

```diff
@@ -16,30 +16,68 @@
 #
 
 load("@vaticle_dependencies//builder/java:rules.bzl", "native_dep_for_host_platform")
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_test")
 
 def typedb_java_test(name, server_mac_artifact, server_linux_artifact, server_windows_artifact,
                       console_mac_artifact = None, console_linux_artifact = None, console_windows_artifact = None,
                       native_libraries_deps = [], deps = [], classpath_resources = [], data = [], args = [], **kwargs):
+
     native_server_artifact_paths, native_server_artifact_labels = native_artifact_paths_and_labels(
-        server_mac_artifact, server_linux_artifact, server_windows_artifact
-    )
-    native_console_artifact_paths, native_console_artifact_labels = [], []
-    if console_mac_artifact and console_linux_artifact and console_windows_artifact:
-        native_console_artifact_paths, native_console_artifact_labels = native_artifact_paths_and_labels(
+           server_mac_artifact, server_linux_artifact, server_windows_artifact
+       )
+
+    native_console_artifact_paths, native_console_artifact_labels = native_console_artifact_paths_and_labels(
             console_mac_artifact, console_linux_artifact, console_windows_artifact
         )
-    native_deps = []
-    for dep in native_libraries_deps:
-        native_deps = native_deps + native_dep_for_host_platform(dep)
+
+    native_deps = add_native_libraries_deps(native_libraries_deps)
```

Comment:
You're right, this was poorly named. We're taking the passed `native_libraries_deps` and turning them into the dependencies that they specify.

Changing the name `native_libraries_deps` in the method signature of `typedb_java_test` could be perilous, see https://github.com/search?p=1&q=org%3Avaticle+native_libraries_deps&type=Code

As such, I've gone with 
```py
native_dependencies = get_native_dependencies(native_libraries_deps)
```
I read this as 'we are getting the native dependencies from the specifications of the platform-specific dependencies we are including with regards to our target platform'.

This is a pretty thorny one, so let me know what you think.

Reply:
Okay, I think we may need to address the heart of this problem, which is that `native_libraries_deps` itself is an awkward, ambiguous name. This will be a fairly wide-reaching refactor.

Reply:
Would renaming the variable immediately work as a stop-gap, or should we leave it as is?

I've raised an issue for this:
- https://github.com/vaticle/typedb-common/issues/130

Reply:
Let's stick with the imperfect name and merge this PR. Finding a perfect name would be impossible in an imperfect environment.